docker-entrypoint.sh

```diff
@@ -1,61 +1,366 @@
+# If neither is set, we need to print a migration info and default to the example-ors-config.env file.
+# Check if ors_config_location is a .json file and exists
+if [[ "${ors_config_location}" = *.yml ]] && [[ -f "${ors_config_location}" ]]; then
+  success "Using yml config: ${ors_config_location}"
+elif [[ "${ors_config_location}" = *.json ]] && [[ -f "${ors_config_location}" ]]; then
+  success "Using json config: ${ors_config_location}"
+  # Print the above warning message in individual warning calls
+  warning ".json configurations are deprecated and will be removed in the future."
+  print_migration_info="true"
+elif [[ -f ${ORS_HOME}/config/ors-config.yml ]]; then
+    success "Using existing ${ORS_HOME}/config/ors-config.yml"
+    ors_config_location="${ORS_HOME}/config/ors-config.yml"
+else
+  warning "No config file found. Copying /example-ors-config.yml to ${ORS_HOME}/config/ors-config.yml"
+  warning "To adjust your config edit ors-config.yml in your 'config' docker volume or use environment variable configuration."
+  update_file "${ORS_HOME}/config/ors-config.yml" "/example-ors-config.yml"
+  ors_config_location="${ORS_HOME}/config/ors-config.yml"
 fi
 
-echo "### openrouteservice configuration ###"
-# Always overwrite the example config in case another one is present
-cp -f "${ors_base}/tmp/ors-config.yml" "${ors_base}/ors-conf/ors-config-example.yml"
-# Check for old .json configs
-JSON_FILES=$(ls -d -- "${ors_base}/ors-conf/"*.json 2>/dev/null)
-if [ -n "$JSON_FILES" ]; then
-    echo "Old .json config found. They're deprecated and will be replaced in ORS version 8."
-    echo "Please migrate to the new .yml example."
+# Get relevant configuration information from the .yml or .json file
+if [[ -z "${ors_engine_graphs_root_path}" ]] && [[ "${config_location}" = *.yml ]]; then
+  ors_engine_graphs_root_path=$(extract_config_info "${ors_config_location}" '.ors.engine.graphs_root_path')
 fi
-# No config found. Use the base config
-if [ ! -f "${ors_base}/ors-conf/ors-config.yml" ]; then
-  echo "Copy ors-config.yml"
-  cp -f "${ors_base}/tmp/ors-config.yml" "${ors_base}/ors-conf/ors-config.yml"
+
+if [[ -z "${ors_engine_graphs_root_path}" ]] && [[ "${config_location}" = *.json ]]; then
+  ors_engine_graphs_root_path=$(extract_config_info "${ors_config_location}" '.ors.services.routing.profiles.default_params.graphs_root_path')
 fi
 
-if [ ! -f "${ors_base}/ors-core/data/osm_file.pbf" ]; then
-  echo "Copy osm_file.pbf"
-  cp -f "${ors_base}/tmp/osm_file.pbf" "${ors_base}/ors-core/data/osm_file.pbf"
+if [[ -z "${ors_engine_source_file}" ]] && [[ "${ors_config_location}" = *.yml ]]; then
+  ors_engine_source_file=$(extract_config_info "${ors_config_location}" '.ors.engine.source_file')
+fi
+
+if [[ -z "${ors_engine_source_file}" ]] && [[ "${ors_config_location}" = *.json ]]; then
+  ors_engine_source_file=$(extract_config_info "${ors_config_location}" '.ors.services.routing.sources[0]')
```

Comment:
We could refactor this to reflect the:
```
does it exist?
   either get from yml
   or get from json
```
structure

Reply:
@jhaeu please revisit this section. You've reworked this quite a lot.